tests/integration/libssh.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash -e
+# Copyright (C) 2024 Ondrej Moris <omoris@redhat.com>
+# SPDX-License-Identifier: Apache-2.0
+
+source "../helpers.sh"
+
+BASEDIR=$PWD
+WORKDIR=$(mktemp -d)
+PKCS11_DEBUG_FILE="${WORKDIR}/pkcs11-libssh-test.log"
+
+install_dependencies()
+{
+    title PARA "Install dependencies" 
+    
+    dnf install -y --skip-broken cmake libcmocka libcmocka-devel softhsm \
+      nss-tools gnutls-utils p11-kit p11-kit-devel p11-kit-server opensc \
+      softhsm-devel socket_wrapper nss_wrapper uid_wrapper pam_wrapper \
+      priv_wrapper openssh-server zlib-devel git autoconf autoconf-archive \
+      automake libtool openssl-devel gcc g++ libcmocka-devel 
+}
+
+pkcs11_provider_setup()
```

Comment:
Two comments here:
- why do you re-checkout and build pkcs11-provider when the git hub action can do that  (in a cached way) for you already ?

- a few of these steps seem to be repeated in the various scripts, wouldn't ii make sense to have a common.sh library imported in the various scripts and execute from there so that we reduce the maintenance cost later ?

Reply:
This pkcs11_provider_setup() is skipped in GH Actions, it is only intended to run when the test is executed manually. When it fails in CI it is useful for debugging to be able to run the test locally, that is now possible. Some code de-duplication might be useful though.

Reply:
I simplified the workflow code a lot with matrix strategy.